src/gear.rs

```diff
@@ -39,6 +39,13 @@ impl Engine for Gear {
     fn digest(&self) -> u32 {
         self.digest.0
     }
+
+    #[inline(always)]
+    fn reset(&mut self) {
+        let chunk_bits = self.chunk_bits;
+        *self = Default::default();
+        self.chunk_bits = chunk_bits;
```

Comment:
I think you can just do `*self = Gear { chunk_bits: self.chunk_bits, ..Default::default() }`